build/syntax.go

```diff
@@ -108,6 +110,14 @@ func (f *File) DisplayPath() string {
 	return f.Path
 }
 
+// Full label of the file, e.g. "//package:file.bzl"
+func (f *File) FullLabel() string {
+	if f.Pkg == "" {
+		return fmt.Sprintf("//%s", f.Label)
```

Comment:
Why do we need this?
Shouldn't the label show up as `//:foo.bzl`?

Reply:
Actually what's more useful is a canonical path of a file (relative to the workspace root, with forward slashes only and no colon for package boundaries. It can be used in warning messages for referring a file (e.g. "the function is defined at //path/to/file.bzl"), but also for diagnostics (e.g. for resolving visibility of .bzl files).

I'll replace the function with CanonicalPath() that returns the same but with slashes instead of colons.

There's a `Path` field in the File struct, but it contains filesystem-specific slashes and may be not relative to the workspace root, e.g. if the workspace root is not known.

Reply:
Maybe it shouldn't start with `//` if it's not a label?

Reply:
Also the name of the function is misleading.